lib/src/webdav/client.dart

```diff
@@ -43,6 +47,28 @@ class WebDavClient {
     return '$_baseUrl/remote.php/dav/$path';
   }
 
+  Future<String> _addFilesPath(String path) async {
+    if (path.startsWith('/files/')) {
+      _useBackwardsCompatiblePaths = true;
+      return path;
+    }
+    if (path.startsWith('/')) {
+      path = path.substring(1, path.length);
+    }
+    _username =
+        _username ?? (await UserClient(_baseUrl, _network).getUser()).id;
```

Comment:
Why not create a factory method and retrieve the `username` there during initialization? Would make the rest of the code easier to read.

Reply:
Future in factorys or any constructors don't work

Reply:
Ah, right, there was something. Thx.